test/unittests/capi_test.cpp

```diff
@@ -87,17 +87,55 @@ TEST(capi, find_exported_function)
     ASSERT_NE(module, nullptr);
 
     uint32_t func_idx;
-    EXPECT_TRUE(fizzy_find_exported_function(module, "foo", &func_idx));
+    EXPECT_TRUE(fizzy_find_exported_function_index(module, "foo", &func_idx));
     EXPECT_EQ(func_idx, 0);
 
-    EXPECT_FALSE(fizzy_find_exported_function(module, "bar", &func_idx));
-    EXPECT_FALSE(fizzy_find_exported_function(module, "g1", &func_idx));
-    EXPECT_FALSE(fizzy_find_exported_function(module, "tab", &func_idx));
-    EXPECT_FALSE(fizzy_find_exported_function(module, "mem", &func_idx));
+    EXPECT_FALSE(fizzy_find_exported_function_index(module, "bar", &func_idx));
+    EXPECT_FALSE(fizzy_find_exported_function_index(module, "g1", &func_idx));
+    EXPECT_FALSE(fizzy_find_exported_function_index(module, "tab", &func_idx));
+    EXPECT_FALSE(fizzy_find_exported_function_index(module, "mem", &func_idx));
 
     fizzy_free_module(module);
 }
 
+TEST(capi, find_exported_function)
+{
+    /* wat2wasm
+    (module
+      (func $f (export "foo") (result i32) (i32.const 42))
+      (global (export "g1") i32 (i32.const 42))
+      (table (export "tab") 10 30 anyfunc)
+      (memory (export "mem") 1 2)
+    )
+    */
+    const auto wasm = from_hex(
+        "0061736d010000000105016000017f0302010004050170010a1e0504010101020606017f00412a0b0718040366"
+        "6f6f00000267310300037461620100036d656d02000a06010400412a0b");
+
+    auto module = fizzy_parse(wasm.data(), wasm.size());
+    ASSERT_NE(module, nullptr);
+
+    auto instance = fizzy_instantiate(module, nullptr, 0, nullptr, nullptr, nullptr, 0);
+    ASSERT_NE(instance, nullptr);
+
+    FizzyExternalFunction function;
+    ASSERT_TRUE(fizzy_find_exported_function(instance, "foo", &function));
+    EXPECT_EQ(function.type.inputs_size, 0);
+    EXPECT_EQ(function.type.output, FizzyValueTypeI32);
+    EXPECT_NE(function.context, nullptr);
+    ASSERT_NE(function.function, nullptr);
+    EXPECT_THAT(function.function(function.context, instance, nullptr, 0), CResult(42));
+
+    fizzy_free_exported_function(&function);
```

Comment:
Since you state `fizzy_free_exported_function(nullptr)` works, can you add it somewhere?

Reply:
It was a mistake, it's not allowed, will change the comment.

Reply:
Changed.